sources/web/datalab/polymer/modules/jupyter-file-manager/jupyter-file-manager.ts

```diff
@@ -18,64 +18,174 @@
  * these APIs to help with type checking.
  */
 
+class JupyterFile extends DatalabFile {
```

Comment:
It might also be worth defining an interface type for the upstream jupyter file. I'm fine with postponing that to a future PR.